src/Arcus.WebApi.Security/Authentication/Certificates/CertificateAuthenticationFilter.cs

```diff
@@ -33,71 +38,95 @@ public async Task OnAuthorizationAsync(AuthorizationFilterContext context)
             Guard.For<ArgumentException>(() => context.HttpContext.RequestServices is null, "Invalid action context given without any HTTP request services");
 
             IServiceProvider services = context.HttpContext.RequestServices;
-            ILogger logger = GetLoggerOrDefault(services);
+            ILogger logger = services.GetLoggerOrDefault<CertificateAuthenticationFilter>();
 
-            X509Certificate2 clientCertificate = GetOrLoadClientCertificateFromRequest(context.HttpContext, logger);
-            if (clientCertificate == null)
+            var validator = services.GetService<CertificateAuthenticationValidator>();
+            if (validator is null)
             {
-                
-                logger.LogWarning(
-                    "No client certificate was specified in the HTTP request while this authentication filter "
-                    + "requires a certificate to validate on the configured validation requirements");
-
-                context.Result = new UnauthorizedResult();
+                throw new KeyNotFoundException(
+                    $"No configured {nameof(CertificateAuthenticationValidator)} instance found in the request services container. "
+                    + "Please configure such an instance (ex. in the Startup) of your application");
             }
-            else
+
+            if (TryGetClientCertificateFromRequest(context.HttpContext, logger, out X509Certificate2 clientCertificate))
             {
-                var validator = services.GetService<CertificateAuthenticationValidator>();
-                if (validator == null)
+                bool isCertificateAllowed = await validator.IsCertificateAllowedAsync(clientCertificate, services);
+                if (isCertificateAllowed)
                 {
-                    throw new KeyNotFoundException(
-                        $"No configured {nameof(CertificateAuthenticationValidator)} instance found in the request services container. "
-                        + "Please configure such an instance (ex. in the Startup) of your application");
+                    LogSecurityEvent(logger, LogLevel.Information, "Client certificate in request is considered allowed according to configured validation requirements");
                 }
-
-                bool isCertificateAllowed = await validator.IsCertificateAllowedAsync(clientCertificate, services);
-                if (!isCertificateAllowed)
+                else
                 {
-                    context.Result = new UnauthorizedResult();
+                    LogSecurityEvent(logger, LogLevel.Error, "Client certificate in request is not considered allowed according to the configured validation requirements", HttpStatusCode.Unauthorized);
```

Comment:
While this is indeed an error, Arcus is handling it correctly so not really an "error" from an application perspective. We shouldn't page on-call people for this for example.
```suggestion
                    LogSecurityEvent(logger, LogLevel.Trace, "Client certificate in request is not considered allowed according to the configured validation requirements", HttpStatusCode.Unauthorized);
```

Do you agree @fgheysels ?